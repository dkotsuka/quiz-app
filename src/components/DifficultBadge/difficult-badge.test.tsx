import React from 'react'
import { DifficultBadge } from '.'
import {create} from 'react-test-renderer';

describe("DifficultBadge", () => {
    it("easy badge matches snapshot", () => {
        const easyBadge = create(<DifficultBadge level="easy" />)
        expect(easyBadge).toMatchSnapshot()
    })

    it("medium badge matches snapshot", () => {
        const mediumBadge = create(<DifficultBadge level="medium" />)
        expect(mediumBadge).toMatchSnapshot()
    })

    it("hard badge matches snapshot", () => {
        const hardBadge = create(<DifficultBadge level="hard" />)
        expect(hardBadge).toMatchSnapshot()
    })
})