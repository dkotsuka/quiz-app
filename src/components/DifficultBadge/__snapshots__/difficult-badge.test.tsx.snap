// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DifficultBadge easy badge matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d7d8de",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 20,
        "justifyContent": "flex-start",
        "paddingHorizontal": 6,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/full-star.png",
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/empty-star.png",
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/empty-star.png",
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto",
          "fontSize": 12,
        },
        Object {
          "paddingHorizontal": 6,
        },
      ]
    }
  >
    Easy
  </Text>
</View>
`;

exports[`DifficultBadge hard badge matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d7d8de",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 20,
        "justifyContent": "flex-start",
        "paddingHorizontal": 6,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/full-star.png",
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/full-star.png",
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/full-star.png",
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto",
          "fontSize": 12,
        },
        Object {
          "paddingHorizontal": 6,
        },
      ]
    }
  >
    Hard
  </Text>
</View>
`;

exports[`DifficultBadge medium badge matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d7d8de",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 20,
        "justifyContent": "flex-start",
        "paddingHorizontal": 6,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/full-star.png",
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/full-star.png",
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/empty-star.png",
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto",
          "fontSize": 12,
        },
        Object {
          "paddingHorizontal": 6,
        },
      ]
    }
  >
    Medium
  </Text>
</View>
`;
